core/trino-main/src/main/java/io/trino/server/testing/TestingTrinoServer.java

```diff
@@ -258,11 +258,12 @@ private TestingTrinoServer(
 
         properties = new HashMap<>(properties);
         int httpPort = parseInt(firstNonNull(properties.remove("http-server.http.port"), "0"));
+        String catalogManagement = firstNonNull(properties.remove("catalog.management"), "dynamic");
```

Comment:
I don't think we need to remove them and re-add them ? Instead can we handle in `builder` side to inject a custom catalog management ? Like how we handle for other modules ?

Reply:
Do you mean allowing the custom properties to take precedence over the default ones? In other words, are you suggesting changing from

```java
                .putAll(properties)
                .put("catalog.management", "dynamic")
```

to:

```java
                .put("catalog.management", "dynamic")
                .putAll(properties)
```
?

If that's the case, it will not work because the builder will throw an exception saying that the keys are duplicated.

Reply:
No I meant - What if we could add something like this to the `Builder`

```
        public Builder setCatalogMangerKind(CatalogManagerConfig.CatalogMangerKind catalogMangerKind)
        {
            this.catalogMangerKind = catalogMangerKind;
            return this;
        }
```

And pass this enum to the ctor and update the `catalog.management` accordingly ?

Reply:
Thanks for the explanation. I applied your suggestion.